pkg/tenancy/config_test.go

```diff
@@ -0,0 +1,85 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package tenancy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTenantAllowed(t *testing.T) {
+	incomingTenant := "tenant-a"
+	emptyTenant := ""
+	tcs := []struct {
+		name   string
+		c      AuthConfig
+		forbid bool
+	}{
+		{
+			name: "allow all tenants",
+			c:    &AllowAllTenantsConfig{nonTenants: true},
+		},
+		{
+			name: "allow valid tenants only",
+			c: &selectiveConfig{
+				validTenantsMap: map[string]struct{}{
+					"tenant-a": {},
+					"tenant-b": {},
+				},
+			},
+		},
+		{
+			name: "forbid tenants",
+			c: &selectiveConfig{
+				validTenantsMap: map[string]struct{}{
+					"tenant-b": {},
+				},
+			},
+			forbid: true,
+		},
+	}
+	for _, tc := range tcs {
+		c := tc.c
+		if tc.forbid {
+			require.False(t, c.IsTenantAllowed(incomingTenant), tc.name)
+		} else {
+			require.True(t, c.IsTenantAllowed(incomingTenant), tc.name)
+		}
```

Comment:
I'm not big on using require but these if else statements negate the purpose of using it at all.

Why not:
```suggestion
		require.Equal(t, c.IsTenantAllowed(incomingTenant), !tc.forbid)
```
Or even, changing `forbid` to `allowed` so we don't have negation and it reads much better.